src/main/java/core/basesyntax/handlers/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.handlers;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.model.Fruit;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private StorageDao storageDao;
+
+    public PurchaseOperationHandler(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void doOperation(Fruit fruit, Integer quantity) {
+        Integer initialQuantity = storageDao.get(fruit);
+        if (initialQuantity == null || initialQuantity <= quantity) {
```

Comment:
```suggestion
        if (initialQuantity == null || initialQuantity < quantity) {
```
думаю тут краще прибрати знак рівності, бо якщо в мапі є 10 фруктів, то ми ж можемо купити всі 10